src/classes/RD2_DataMigration_TEST.cls

```diff
@@ -0,0 +1,187 @@
+/*
+    Copyright (c) 2019 Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group Recurring Donations
+* @description Unit Test for the data migration batch job
+*
+*/
+@IsTest
+private class RD2_DataMigration_TEST {
+
+    /**
+     * @description Verifies that the job should run with zero records processed when
+     * Enhanced Recurringn Donations is not enabled
+     */
+    @IsTest
+    private static void shouldLaunchBatchJobWithNothingProcessed() {
+        Contact contact = UTIL_UnitTestData_TEST.getContact();
+        insert contact;
+
+        npe03__Recurring_Donation__c rd = getNewRecurringDonationLegacyFormat(contact.Id);
+        rd.Id = null;
+        insert rd;
+
+        Exception jobException;
+        try {
+            Test.startTest();
+            RD2_DataMigration_BATCH batchJob = new RD2_DataMigration_BATCH();
+            Database.executeBatch(batchJob);
+            Test.stopTest();
+        } catch (Exception ex) {
+            jobException = ex;
+        }
+
+        System.assertNotEquals(null, jobException, 'The batch job did not throw an exception');
```

Comment:
Let's specify expected behaviour in the assert help message. For example: `Batch job should generate an exception.`.